tests/unit/style/properties/serialization.rs

```diff
@@ -0,0 +1,828 @@
+        use style::properties::longhands::transition_timing_function::computed_value::TransitionTimingFunction;
+        use style::values::specified::Time as TimeContainer;
+
+        let property_name = DeclaredValue::Value(
+            PropertyContainer(vec![TransitionProperty::MarginLeft])
+        );
+
+        let duration = DeclaredValue::Value(
+            DurationContainer(vec![TimeContainer(3f32)])
+        );
+
+        let delay = DeclaredValue::Value(
+            DurationContainer(vec![TimeContainer(4f32)])
+        );
+
+        let timing_function = DeclaredValue::Value(
+            TimingContainer(vec![
+                TransitionTimingFunction::CubicBezier(Point2D::new(0f32, 5f32), Point2D::new(5f32, 10f32))
+            ])
+        );
+
+        let mut properties = Vec::new();
+
+        properties.push(PropertyDeclaration::TransitionProperty(property_name));
+        properties.push(PropertyDeclaration::TransitionDelay(delay));
+        properties.push(PropertyDeclaration::TransitionDuration(duration));
+        properties.push(PropertyDeclaration::TransitionTimingFunction(timing_function));
+
+        let serialization = shorthand_properties_to_string(properties);
+        assert_eq!(serialization, "transition: margin-left 3s cubic-bezier(0, 5, 5, 10) 4s;");
+    }
+
+    #[test]
+    fn flex_should_serialize_all_available_properties() {
+        use style::values::specified::Number as NumberContainer;
+        use style::values::specified::Percentage as PercentageContainer;
+
+        let mut properties = Vec::new();
+
+        let grow = DeclaredValue::Value(NumberContainer(2f32));
+        let shrink = DeclaredValue::Value(NumberContainer(3f32));
+        let basis = DeclaredValue::Value(
+            LengthOrPercentageOrAutoOrContent::Percentage(PercentageContainer(0.5f32))
+        );
+
+        properties.push(PropertyDeclaration::FlexGrow(grow));
+        properties.push(PropertyDeclaration::FlexShrink(shrink));
+        properties.push(PropertyDeclaration::FlexBasis(basis));
+
+        let serialization = shorthand_properties_to_string(properties);
+        assert_eq!(serialization, "flex: 2 3 50%;");
+    }
+
+    #[test]
+    fn flex_flow_should_serialize_all_available_properties() {
+        use style::properties::longhands::flex_direction::computed_value::T as FlexDirection;
+        use style::properties::longhands::flex_wrap::computed_value::T as FlexWrap;
+
+        let mut properties = Vec::new();
+
+        let direction = DeclaredValue::Value(FlexDirection::row);
+        let wrap = DeclaredValue::Value(FlexWrap::wrap);
+
+        properties.push(PropertyDeclaration::FlexDirection(direction));
+        properties.push(PropertyDeclaration::FlexWrap(wrap));
+
+        let serialization = shorthand_properties_to_string(properties);
+        assert_eq!(serialization, "flex-flow: row wrap;");
+    }
+
+    // TODO: Populate Atom Cache for testing so that the font shorthand can be tested
```

Comment:
I do not remember off the top of my head now, but Atom::from was causing panic, maybe because it was expecting to find something in the cache that wasn't there?